python/hsfs/client/external.py

```diff
@@ -114,6 +115,44 @@ def __init__(
             self._key_store_path = _spark_session.conf.get(
                 "spark.hadoop.hops.ssl.keystore.name"
             )
+            self.validate_spark_configuration(_spark_session)
```

Comment:
The validation should happen before we try to access the properties in the line above.